src/components/Home/Post.tsx

```diff
@@ -0,0 +1,98 @@
+import { Avatar, Paper } from '@mui/material';
+import { blue, deepPurple } from '@mui/material/colors';
+import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
+import ThumbUpIcon from '@mui/icons-material/ThumbUp';
+import ThumbUpOutlinedIcon from '@mui/icons-material/ThumbUpOutlined';
+import ModeCommentOutlinedIcon from '@mui/icons-material/ModeCommentOutlined';
+import IosShareOutlinedIcon from '@mui/icons-material/IosShareOutlined';
+import { Post } from '../types/Post';
+import { useState } from 'react';
+
+type PostProps = {
+  post: Post;
+  changePost: (inputPost: Post) => void;
+}
+
+const Post = ({post, changePost}: PostProps) => {
+  const [currentPost, setCurrentPost] = useState(post);
+  const likeChange = () => {
+    const temp = currentPost;
+      if (currentPost.youLiked) {
+        temp.youLiked = false;
+        temp.Liked = temp.Liked - 1;
+      }
+      else {
+        temp.youLiked = true;
+        temp.Liked = temp.Liked + 1;
+      }
+      setCurrentPost(temp);
+      changePost(temp);
+  }
+    return (
+    <Paper elevation={2} sx={{ height: "fit-content", width: "70%" }}>
+      <div style={{ display: "flex", flexDirection: "column", padding: 15, gap: 20}}>
```

Comment:
This DOM element with similar style has been repeated for several times, consider encapsulating it as a reusable component.

e.g. 
```
export const RowComponent = ({style, children}) => {
return (<div style={{ display: "flex", flexDirection: "column", padding: 15, gap: 20, ...style}}>
{children}
</div>
)
}
```

Reply:
got it